packages/material-ui-lab/src/TreeView/descendants.js

```diff
@@ -0,0 +1,217 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+
+/** Credit: https://github.com/reach/reach-ui/blob/86a046f54d53b6420e392b3fa56dd991d9d4e458/packages/descendants/README.md
+ *  Modified slightly to suit our purposes.
```

Comment:
Modifications are to allow the parentId to be read through this context. If we wanted to use the source we could use a second context.